packages/eas-cli/src/commands/env/unlink.ts

```diff
@@ -40,57 +45,106 @@ export default class EnvironmentVariableUnlink extends EasCommand {
+
+    if (!nonInteractive && !unlinkEnvironments) {
+      const selectedEnvironments =
+        (selectedVariable.linkedEnvironments ?? []).length > 0
+          ? selectedVariable.linkedEnvironments
+          : selectedVariable.environments;
+      const environments = await promptVariableEnvironmentAsync({
+        nonInteractive,
+        multiple: true,
+        selectedEnvironments: selectedEnvironments ?? [],
+      });
+      explicitSelect = true;
+      unlinkEnvironments = Object.values(EnvironmentVariableEnvironment).filter(
+        env => !environments.includes(env)
+      );
+    }
 
-    if (!unlinkedVariable) {
-      throw new Error(
-        `Could not unlink variable with name ${selectedVariable.name} from project ${projectDisplayName}`
+    if (!unlinkEnvironments) {
+      await EnvironmentVariableMutation.unlinkSharedEnvironmentVariableAsync(
+        graphqlClient,
+        selectedVariable.id,
+        projectId
       );
+      Log.withTick(
+        `Unlinked variable ${chalk.bold(selectedVariable.name)} from project ${chalk.bold(
+          projectDisplayName
+        )} in ${selectedVariable.environments?.join(', ').toLocaleLowerCase()}.`
+      );
+      return;
     }
 
-    Log.withTick(
-      `Unlinked variable ${chalk.bold(unlinkedVariable.name)} from the project ${chalk.bold(
-        projectDisplayName
-      )}.`
-    );
+    for (const environment of Object.values(EnvironmentVariableEnvironment)) {
+      try {
+        if (
+          selectedVariable.linkedEnvironments?.includes(environment) !==
+          unlinkEnvironments.includes(environment)
+        ) {
+          if (!explicitSelect && unlinkEnvironments.includes(environment)) {
+            Log.withTick(
+              `Variable ${chalk.bold(
+                selectedVariable.name
+              )} is already unlinked from ${environment.toLocaleLowerCase()}.`
+            );
+          }
+          continue;
+        }
+        if (unlinkEnvironments.includes(environment)) {
+          await EnvironmentVariableMutation.unlinkSharedEnvironmentVariableAsync(
+            graphqlClient,
+            selectedVariable.id,
+            projectId,
+            environment
+          );
+          Log.withTick(
+            `Unlinked variable ${chalk.bold(selectedVariable.name)} from project ${chalk.bold(
+              projectDisplayName
+            )} in ${environment.toLocaleLowerCase()}.`
+          );
+        } else if (explicitSelect) {
+          await EnvironmentVariableMutation.linkSharedEnvironmentVariableAsync(
+            graphqlClient,
+            selectedVariable.id,
+            projectId,
+            environment
+          );
+          Log.withTick(
+            `Linked variable ${chalk.bold(selectedVariable.name)} to project ${chalk.bold(
+              projectDisplayName
+            )} in ${environment.toLocaleLowerCase()}.`
+          );
+        }
+      } catch (err: any) {
+        Log.warn(err.message);
+      }
```

Comment:
I don't think we should be catching these 🤔 What do you think?

Reply:
Yep, you're right